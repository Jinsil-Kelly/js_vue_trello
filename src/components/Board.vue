<template>
  <div>
    <div>{{ bId }}</div>
    <!--<hr/>-->
    <!--<router-view></router-view>-->
  </div>
</template>

<script>
export default {
  created() {
    this.bId = this.$route.params.bId;
  },
  data() {
    return {
      bId: 0
    };
  },
  props: {
    msg: String
  }
};
</script>

<style lang="scss"></style>
