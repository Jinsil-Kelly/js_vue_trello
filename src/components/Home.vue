<template>
  <div>Home {{ msg }}</div>
</template>

<script>
export default {
  props: {
    msg: String
  }
};
</script>

<style lang="scss"></style>
