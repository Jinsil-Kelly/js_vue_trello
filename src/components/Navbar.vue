<template>
  <nav>
    <router-link to="/">Home</router-link>

    <router-link to="/login">Login</router-link>
    <router-link to="/b/">Board</router-link>
  </nav>
</template>

<script>
export default {
  name: "Navbar"
};
</script>

<style scoped></style>
