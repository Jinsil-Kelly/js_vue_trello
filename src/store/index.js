import Vue from "vue";
import Vuex from "vuex";
import board from "@/store/modules/board";
import auth from "@/store/modules/auth";

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    auth,
    board
  }
});
