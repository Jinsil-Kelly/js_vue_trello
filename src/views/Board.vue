<template>
  <div v-if="loading">I am loading....</div>
  <div v-else>
    <div>{{ bId }}</div>

    <router-link :to="`/b/${bId}/c/1`">Card1</router-link>
    <router-link :to="`/b/${bId}/c/2`">Card2</router-link>
    <hr />
    <router-view></router-view>
  </div>
</template>

<script>
export default {
  data() {
    return { bId: 0, loading: true };
  },
  created() {
    return (this.bId = this.$route.params.bId);
  }
};
</script>
