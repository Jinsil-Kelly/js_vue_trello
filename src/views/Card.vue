<template>
  <Modal class="modal-card">
    <div slot="header">
      <EditCardHeader :title="card.title" />
    </div>
    <div slot="body">
      <h3>Description</h3>
      <textarea
        class="form-control"
        cols="30"
        rows="3"
        placeholder="Add a more detailed description..."
        :readonly="!toggleDesc"
        @click="toggleDesc = true"
        @blur="onBlurDesc"
        v-model="card.description"
        ref="inputDesc"
      ></textarea>
    </div>
    <div slot="footer"></div>
  </Modal>
</template>

<script>
import { createNamespacedHelpers, mapGetters } from "vuex";
const { mapActions } = createNamespacedHelpers("card");
import Modal from "@/components/Modal.vue";
import EditCardHeader from "@/components/EditCardHeader.vue";

export default {
  components: { Modal, EditCardHeader },
  data() {
    return {
      toggleTitle: false,
      toggleDesc: false
    };
  },
  computed: {
    ...mapGetters("board", ["board"]),
    ...mapGetters("card", ["card"])
  },
  created() {
    // this.fetch()
    this.FETCH_CARD({ id: this.$route.params.cId });
  },
  methods: {
    ...mapActions(["FETCH_CARD", "UPDATE_CARD"]),
    // fetch(){
    //   this.FETCH_CARD({ id:this.$route.params.cId });
    // },
    onClose() {
      this.$router.push(`/b/${this.board.id}`);
    },
    onBlurTitle() {
      this.toggleTitle = false;
      const title = this.$refs.inputTitle.value.trim();
      if (!title) return;
      this.UPDATE_CARD({
        id: this.card.id,
        title,
        bId: this.$route.params.bId
      }).then(() => this.FETCH_CARD({ id: this.$route.params.cId }));
    },
    onBlurDesc() {
      this.toggleDesc = false;
      const description = this.$refs.inputDesc.value.trim();
      this.UPDATE_CARD({
        id: this.card.id,
        description,
        bId: this.$route.params.bId
      });
    }
  }
};
</script>

<style>
.modal-card .modal-container {
  min-width: 300px;
  max-width: 800px;
  width: 60%;
}
.modal-card-header-title {
  padding-right: 30px;
}
.modal-close-btn {
  position: absolute;
  top: 0px;
  right: 0px;
  font-size: 24px;
  text-decoration: none;
}
.modal-card-header {
  position: relative;
}
</style>
