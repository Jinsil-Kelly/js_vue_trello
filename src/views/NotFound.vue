<template>
  <div class="home">
    <NotFound msg="I am NotFound Page" />
  </div>
</template>

<script>
// @ is an alias to /src
import NotFound from "@/components/NotFound.vue";

export default {
  components: {
    NotFound
  }
};
</script>
