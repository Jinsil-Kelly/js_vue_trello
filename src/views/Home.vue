<template>
  <div class="home">
    <Home msg="Welcome to Jinsil Home" />
    <router-link to="/b/1">Board1</router-link>
    <router-link to="/b/2">Board2</router-link>
  </div>
</template>

<script>
// @ is an alias to /src
import Home from "@/components/Home.vue";

export default {
  name: "home",
  components: {
    Home
  }
};
</script>
