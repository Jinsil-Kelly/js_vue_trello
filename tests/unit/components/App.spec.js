import { shallowMount, mount, createLocalVue } from "@vue/test-utils";
import App from "@/App.vue";
import Router from "vue-router";
import Navbar from "@/components/Navbar.vue";
import Home from "@/views/Home.vue";
import Login from "@/views/Login.vue";
import NotFound from "@/views/NotFound.vue";
import Board from "@/views/Board.vue";
import Card from "@/views/Card.vue";
import store from "@/store";
import Vuex from "vuex";
import Loader from "@/components/Loader";

const localVue = createLocalVue();
localVue.use(Router);
localVue.use(Vuex);

describe("App", () => {
  let isAuth;
  const requireAuth = (to, from, next) => {
    const loginPath = `/login?rPath=${encodeURIComponent(to.path)}`;
    isAuth ? next() : next(loginPath);
  };
  const routes = [
    {
      path: "/",
      name: "home",
      component: Home,
      beforeEnter: requireAuth
    },
    { path: "/login", component: Login },
    {
      path: "/b/:bId",
      component: Board,
      name: "board",
      beforeEnter: requireAuth,

      children: [{ path: "c/:cId", component: Card }]
    },
    {
      path: "/about",
      name: "about",
      component: () =>
        import(/* webpackChunkName: "about" */ "./views/About.vue")
    },
    {
      path: "*",
      component: NotFound
    }
  ];

  const router = new Router({ routes });
  const build = () => {
    const wrapper = mount(App, {
      localVue,
      router,
      store
    });
    return {
      wrapper
    };
  };

  it("renders the component", () => {
    // arrange
    const wrapper = shallowMount(App, {
      localVue,
      router,
      store
    });
    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components - Navbar", () => {
    // arrange
    const wrapper = shallowMount(App, {
      localVue,
      router,
      store
    });
    // assert
    expect(wrapper.find(Navbar).exists()).toBe(true);
    expect(wrapper.findAll(".container").length).toEqual(2);
  });

  it("renders a child component via routing when a user is logged in and path is '/'", () => {
    isAuth = true;
    const { wrapper } = build();
    router.push("/");
    expect(wrapper.find(Login).exists()).toBe(false);
    expect(wrapper.find(Home).exists()).toBe(true);
    expect(wrapper.find(Board).exists()).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders a child component via routing when a user is logged in and path is '/b/1", () => {
    isAuth = true;
    const { wrapper } = build();
    router.push("/b/1");
    expect(wrapper.find(Login).exists()).toBe(false);
    expect(wrapper.find(Home).exists()).toBe(false);
    expect(wrapper.find(Board).exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders a child component via routing when a user is not logged in", () => {
    isAuth = false;
    const { wrapper } = build();
    router.push("/");
    expect(wrapper.find(Login).exists()).toBe(true);
    expect(wrapper.find(Home).exists()).toBe(false);
    expect(wrapper.find(Board).exists()).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders Loader Component when data is loading", () => {
    let getters = {
      loading: () => true
    };
    const mockStore = new Vuex.Store({
      getters
    });
    const wrapper = shallowMount(App, { store: mockStore, localVue });
    getters.loading();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(Loader).exists()).toBe(true);
    expect(wrapper.findAll(".container").at(0).element.style.display).toBe("");
    expect(wrapper.findAll(".container").at(1).element.style.display).toBe(
      "none"
    );
  });

  it("does not render Loader Component when loading data is done", () => {
    let getters = {
      loading: () => false
    };
    const mockStore = new Vuex.Store({
      getters
    });
    const wrapper = shallowMount(App, { store: mockStore, localVue });
    getters.loading();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(Loader).exists()).toBe(true);
    expect(wrapper.findAll(".container").at(0).element.style.display).toBe(
      "none"
    );
    expect(wrapper.findAll(".container").at(1).element.style.display).toBe("");
  });
});
