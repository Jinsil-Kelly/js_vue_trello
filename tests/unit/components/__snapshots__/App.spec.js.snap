// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the component 1`] = `
<div id="app" class="container">
  <navbar-stub></navbar-stub>
  <router-view-stub name="default"></router-view-stub>
</div>
`;
